CMakeLists.txt

```diff
@@ -1,8 +1,8 @@
 cmake_minimum_required(VERSION 3.10)
 project(roboteam_ai)
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS} -g -O0")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -Werror -g -O0")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS} -g -O0 -fPIE")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -Werror -g -O0 -fPIE")
```

Comment:
You shouldn't need this. This is pretty bad, as it means your code is position dependent somehow which means you are doing some very shady c++ magic somewhere (without looking at the entire PR.)

Reply:
Fixed it by making NFParam shared instead of static. Turns out static libs don't like Address Space Layout Randomization